### PR TITLE
Add codegen for admin schema

### PR DESCRIPTION
**Pull request by catlee**
### WHY are these changes introduced?

Add codegen for admin schema required by cli-kit to integrate with the [new theme graphQL APIs](https://shopify.dev/changelog/introducing-theme-and-theme-file-management-in-the-admin-graphql-api)

### CODE REVIEW

File: packages/cli-kit/src/public/node/api/admin.ts [hunk 1]
```diff
@@ -21,6 +23,34 @@ export async function adminRequest<T>(query: string, session: AdminSession, vari
...
+  }
+  const opts = {
+    url: adminUrl(session.storeFqdn, apiVersion),
+    api: 'Admin',
+    token: session.token,
+  }
+  const result = graphqlRequestDoc<TResult, TVariables>({...opts, query, variables})
+  return result
+}
+
```
Comment (karreiro) [inline packages/cli-kit/src/public/node/api/admin.ts:53]: I've noticed the `fetchApiVersions` call had the `responseOptions: {handleErrors: false}`. It seems like that impacts how responses are handled and may control how they are [logged](https://github.com/Shopify/cli/blob/a2bb5689b8ad370774af9ed75dcac150d28cda03/packages/cli-kit/src/public/node/api/graphql.ts#L65).

I believe we could keep that in our new API, so I'd update this to something like this:

```suggestion
/**
 * Executes a GraphQL query against the Admin API. Uses typed documents.
 *
 * @param query - GraphQL query to execute.
 * @param session - Shopify admin session including token and Store FQDN.
 * @param variables - GraphQL variables to pass to the query.
 * @param version - API version.
 * @param responseOptions - Control how API responses will be handled.
 * @returns The response of the query of generic type <TResult>.
 */
export async function adminRequestDoc<TResult, TVariables extends Variables>(
  query: TypedDocumentNode<TResult, TVariables>,
  session: AdminSession,
  variables?: TVariables,
  version?: string,
  responseOptions?: GraphQLResponseOptions<TResult>,
): Promise<TResult> {
  let apiVersion = version
  if (!version) {
    apiVersion = await fetchLatestSupportedApiVersion(session)
  }
  const opts = {
    url: adminUrl(session.storeFqdn, apiVersion),
    api: 'Admin',
    token: session.token,
  }
  const result = graphqlRequestDoc<TResult, TVariables>({...opts, query, variables, responseOptions})
  return result
}
```

**Review by karreiro — APPROVED** [review]
Thank you for this PR, @catlee! LGTM, I've left only two minor suggestions :)

File: bin/get-graphql-schemas.js [hunk 2]
```diff
@@ -70,14 +75,14 @@ async function fetchFileForSchema(schema, octokit) {
   try {
     // Fetch the file content from the repository
     const {data} = await octokit.repos.getContent({
+      mediaType: { format: "raw" },
```
Comment (shauns) [inline bin/get-graphql-schemas.js:78]: 👍 nice

**Review by shauns — APPROVED** [review]
Have approved, but there's an open question about the right folder for this. Plus Guilherme's comments 👍

File: graphql.config.ts [hunk 3]
```diff
@@ -73,5 +73,6 @@ export default {
     businessPlatformDestinations: projectFactory('business-platform-destinations', 'destinations_schema.graphql'),
     businessPlatformOrganizations: projectFactory('business-platform-organizations', 'organizations_schema.graphql'),
     appDev: projectFactory('app-dev', 'app_dev_schema.graphql'),
+    admin: projectFactory('admin', 'admin_schema.graphql', 'cli-kit'),
```
[the remark this answers is not in the document]
Comment (catlee) [reply graphql.config.ts:76]: I really struggled to find a good place to put it.

Most of the APIs used by themes are defined in `packages/cli-kit/src/public/node/themes/api.ts`, and I couldn't figure out how to have the queries defined in the `theme` package but consumed by the api module in `cli-kit`.

Perhaps that argues for moving all this logic into `theme`?

File: packages/cli-kit/src/public/node/api/admin.ts (hunk 1 above)
Comment (catlee) [reply packages/cli-kit/src/public/node/api/admin.ts:53]: Great catch, thank you!

**Pull request merged**